UnityProject/Assets/Scripts/UI/Items/GUI_Explosive.cs

```diff
@@ -68,62 +67,39 @@ private IEnumerator WaitForProvider()
 			{
 				yield return WaitFor.EndOfFrame;
 			}
-			sbButtons.SetActive(false);
+
+			sbButtons.SetActive(false); //Disable this via code because NetUI doesn't register it from the prefab if it is disabled by default
 			explosiveDevice = Provider.GetComponent<ExplosiveBase>();
-			if (explosiveDevice.DetonateImmediatelyOnSignal == false)
-			{
-				timerCount = explosiveDevice.TimeToDetonate;
-				DisplayTime();
-			}
-			else
-			{
-				timer.Value = "Waiting signal..";
-				status.ElementTmp.color = dangerColor;
-			}
 
+			//Setup variables and graphics
+			timerCount = explosiveDevice.TimeToDetonate;
 			EnsureVisualsAreCorrect();
+			UpdateStatusText();
+			DisplayTime();
 
-			modeToggleButton.Value = explosiveDevice.DetonateImmediatelyOnSignal ? "1" : "0";
 			timerCount = explosiveDevice.TimeToDetonate;
 			timer.Value = DisplayTime();
 			explosiveDevice.GUI = this;
 		}
 
 		public void ArmDevice()
 		{
-			if (explosiveDevice.ExplosiveType == ExplosiveType.SyndicateBomb)
-			{
-				explosiveDevice.IsArmed = sbArmToggleButton.Value == "1";
-			}
-			else
-			{
-				explosiveDevice.IsArmed = armToggleButton.Value == "1";
-			}
-			if ((modeToggleButton.Value == "0" || sbModeToggleButton.Value == "0")
-			    && (armToggleButton.Value == "1" || sbArmToggleButton.Value == "1"))
-			{
-				modeToggleButton.enabled = false;
-				sbModeToggleButton.enabled = false;
-				armToggleButton.enabled = false;
-				sbArmToggleButton.enabled = false;
-				StartCoroutine(explosiveDevice.Countdown());
-				UpdateStatusText();
-				return;
-			}
+			explosiveDevice.IsArmed = true;
+			StartCoroutine(explosiveDevice.Countdown());
 			UpdateStatusText();
 		}
 
 		public void ToggleMode()
 		{
-			if(modeToggleButton.Value == "1" || sbModeToggleButton.Value == "1")
-			{
-				explosiveDevice.ToggleMode(true);
-			}
-			else
+			explosiveDevice.ToggleMode(!explosiveDevice.DetonateImmediatelyOnSignal);
+			signalIcon.SetValueServer(explosiveDevice.DetonateImmediatelyOnSignal ? "0" : "1");
+			UpdateStatusText();
+			if(Peepers.Count == 0) return;
```

Comment:
This statement doesn't add anything.

Reply:
...which one?

Reply:
I assume you mean `if(Peepers.Count == 0) return;` so i just removed it and fixed other stuff.